src/awkward/operations/convert.py

```diff
@@ -2658,12 +2658,11 @@ def popbuffers(array, tpe, buffers):
         elif isinstance(tpe, pyarrow.lib.FixedSizeListType):
             assert tpe.num_buffers == 1
             mask = buffers.pop(0)
-            offsets = ak.layout.Index32(numpy.array([0, tpe.list_size], dtype=np.int32))
             content = popbuffers(array.values, tpe.value_type, buffers)
             if not tpe.value_field.nullable:
                 content = content.content
 
-            out = ak.layout.ListOffsetArray32(offsets, content)
+            out = ak.layout.RegularArray(content, tpe.list_size)
```

Comment:
The FixedSizeListType corresponds to an `ak.layout.RegularArray` for regular-length lists.

Also, these `offsets` only happened to be right for the length-1 example you tested: `tpe.list_size` is the length of all of the fixed-length lists; `offsets` of `np.array([0, N])` would mean the whole array consists of one list whose length is `N`. (That's why it worked when the test case had `len(out) == 1`.)

_If_ we wanted to turn the FixedSizeListType into an `ak.layout.ListOffsetArray64`, then the appropriate `offsets` would have been `np.arange(0, (len(array) + 1)*tpe.list_size, tpe.list_size)`.

Reply:
:facepalm: this is what I was worried about! I misunderstood this to be a fixed size 1D array. Thanks for the catch.